src/WebJobs.Extensions.DurableTask/EndToEndTraceHelper.cs

```diff
@@ -348,22 +380,36 @@ public void ExtensionWarningEvent(string hubName, string functionName, string in
                 LocalSlotName,
                 functionName,
                 instanceId,
-                reason,
+                sanitizedReason,
                 functionType.ToString(),
                 ExtensionVersion,
                 IsReplay: false);
 
             this.logger.LogWarning(
                 "{instanceId}: Function '{functionName} ({functionType})' was rewound. Reason: {reason}. State: {state}. HubName: {hubName}. AppName: {appName}. SlotName: {slotName}. ExtensionVersion: {extensionVersion}. SequenceNumber: {sequenceNumber}.",
-                instanceId, functionName, functionType, reason, FunctionState.Rewound, hubName,
+                instanceId, functionName, functionType, loggerReason, FunctionState.Rewound, hubName,
                 LocalAppName, LocalSlotName, ExtensionVersion, this.sequenceNumber++);
         }
 
+        public void FunctionFailed(
+            string hubName,
+            string functionName,
+            string instanceId,
+            Exception? exception,
+            FunctionType functionType,
+            bool isReplay,
+            int taskEventId = -1)
+        {
+            string sanitizedReason = this.SanitizeException(exception, out string loggerReason, isReplay);
+            this.FunctionFailed(hubName, functionName, instanceId, loggerReason, sanitizedReason, functionType, isReplay, taskEventId);
+        }
+
         public void FunctionFailed(
             string hubName,
             string functionName,
             string instanceId,
             string reason,
+            string sanitizedReason,
```

Comment:
I noticed that `sanitizedReason` is going to ETW while `reason` is going to `ILogger`, but don't we send both of these payloads to Kusto, ETW to DurableFunctionsEvents and ILogger to FunctionsLogs? I know that's not the case for the DTFx tracing code, but I thought the WebJobs `ILogger` logs went to FunctionsLogs.

Reply:
You're right. Throughout this PR, we only sanitize the logs to `DurableFunctionsEvents` and not the ILogger-powered logs to `FunctionsLogs`. This is because the ILogger logs are also sent to the user's Application Insights instance, and I was worried about changing the logging behavior to a user-facing component.

So I'm choosing to delay our decision to sanitize the FunctionsLogs for now so that we can unblock sanitizing the DF Kusto table. Does that seem reasonable?

Reply:
OK, thanks for clarifying. It would be good to confirm what the plan is for sanitizing `FunctionsLogs` just to make sure that the work done in this PR isn't redundant or need to be reversed. Happy to chat about this offline.

Reply:
Let's discuss this offline. But in general I think the clean up done here is unlikely to need to be reversed. If anything, I think it's more likely that it will need to be expanded to include FunctionsLogs as well. I'm just opting to merge a minimal improvement for now.